### PR TITLE
[bug 1588625] Add workerConfig to aws provider as well

### PR DESCRIPTION
**Pull request by imbstack**
This works in concert with taskcluster/taskcluster-worker-runner#51.

This was mostly me guessing so feel free to yell that we should do this differently!

### CODE REVIEW

File: services/worker-manager/src/providers/aws.js [hunk 1]
```diff
@@ -107,6 +89,25 @@ class AwsProvider extends Provider {
...
+        workerGroup: this.providerId,
+        workerConfig: config.launchConfig.workerConfig || {},
+      }));
+      // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html#instancedata-add-user-data
+      // The raw data should be 16KB maximum
+      if (userData.length > 16384) {
+        return await workerPool.reportError({
+          kind: 'creation-error',
+          title: 'User Data is too long',
+          description: 'Try a shorter workerPoolId and/or a shorter rootUrl',
```
Comment (djmitche) [inline services/worker-manager/src/providers/aws.js]: Maybe better advice here is "try removing some workerConfiguration and consider putting it in a secret"?

File: services/worker-manager/src/providers/aws.js [hunk 2]
```diff
@@ -107,6 +89,25 @@ class AwsProvider extends Provider {
         ts.ResourceType === 'instance' ? instanceTags.concat(ts.Tags) : otherTagSpecs.push(ts)
       );
 
+      const userData = Buffer.from(JSON.stringify({
```
Comment (owlishDeveloper) [inline services/worker-manager/src/providers/aws.js]: Why was this moved?

Comment (imbstack) [reply services/worker-manager/src/providers/aws.js]: Because the workerconfig comes from the launchconfigs so it needs to be inside the loop
